app/mailers/devise_notify_mailer.rb

```diff
@@ -36,10 +36,20 @@ def reset_password_instructions(record, token, _opts = {})
 
   def confirmation_instructions(record, token, _opts = {})
     url = "#{user_confirmation_url}?confirmation_token="
+
+    username = record.email
+    if record.confirmable_template == User::CONFIRMABLE_TEMPLATE_ID && (record.unconfirmed_email.present? && record.unconfirmed_email != record.email)
```

Comment:
Is this intended to handle reconfirmations? Might be worth extracting out into it's own method?

Reply:
This is for email changes. Good point

Reply:
better now?